sdk/core/core-amqp/src/errors.ts

```diff
@@ -547,47 +610,62 @@ function isBrowserWebsocketError(err: any): boolean {
   return result;
 }
 
+const rheaPromiseErrors = [
+  // OperationTimeoutError occurs when the service fails to respond within a given timeframe.
+  "OperationTimeoutError",
+
+  // InsufficientCreditError occurs when the number of credits available on Rhea link is insufficient.
+  "InsufficientCreditError",
+
+  // Defines the error that occurs when the Sender fails to send a message.
+  "SendOperationFailedError"
+];
+
 /**
- * Translates the AQMP error received at the protocol layer or a generic Error into a MessagingError.
+ * This utility checks if an error is a retryable error from rhea-promise.
+ * @param err
+ */
+function isRheaPromiseError(err: Error): boolean {
+  return rheaPromiseErrors.indexOf(err.name) !== -1;
+}
+
+/**
+ * Translates the AQMP error received at the protocol layer or a SystemError into a MessagingError.
+ * All other errors are returned unaltered.
  *
  * @param {AmqpError} err The amqp error that was received.
  * @returns {MessagingError} MessagingError object.
  */
-export function translate(err: AmqpError | Error): MessagingError {
-  if ((err as MessagingError).translated) {
+export function translate(err: AmqpError | Error): MessagingError | Error {
+  if ((err as MessagingError).name === "MessagingError") {
     // already translated
     return err as MessagingError;
   }
 
-  let error: MessagingError = err as MessagingError;
-
   // Built-in errors like TypeError and RangeError should not be retryable as these indicate issues
   // with user input and not an issue with the Messaging process.
   if (err instanceof TypeError || err instanceof RangeError) {
-    error.retryable = false;
-    return error;
+    return err;
   }
 
   if (isAmqpError(err)) {
     // translate
     const condition = (err as AmqpError).condition;
     const description = (err as AmqpError).description as string;
```

Comment:
The casting to `AmqpError` above can be removed now that `isAmqpError` is tweaked to ensure that if true, then the type is indeed AmqpError.
Same for a few lines below for `error.info`

Reply:
Updated